hs-bindgen/src-internal/HsBindgen/Frontend/AST/External.hs

```diff
@@ -265,3 +266,9 @@ data TypedefRef =
 isVoid :: Type -> Bool
 isVoid TypeVoid = True
 isVoid _        = False
+
+-- | Arrays can be of known or unknown size.
+isArray :: Type -> Bool
+isArray TypeConstArray{}      = True
+isArray TypeIncompleteArray{} = True
```

Comment:
This may or may not be correct; we don't actually support this yet. We should consider when looking at https://github.com/well-typed/hs-bindgen/issues/377 .

Reply:
Added a TODO here that references #377.